docker/bookkeeper/entrypoint.sh

```diff
@@ -73,6 +73,24 @@ create_bookie_dirs() {
   done
 }
 
+# Create a Bookie ID if this is a newly added bookkeeper pod
+# or read the Bookie ID if a cookie containing this value already exists
+set_bookieid() {
+  IFS=',' read -ra journal_directories <<< $BK_journalDirectories
+  COOKIE="${journal_directories[0]}/current/VERSION"
+  if [ `find ${COOKIE} | wc -l` -gt 0 ]; then
+    # Reading the Bookie ID value from the existing cookie
+    bkHost=`cat ${COOKIE} | grep bookieHost`
```

Comment:
I think this is assuming a text format, but checking the BK code, it looks like more recent versions format it using protobuf:

https://github.com/apache/bookkeeper/blob/master/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java#L195
https://github.com/apache/bookkeeper/blob/master/bookkeeper-proto/src/main/proto/DataFormats.proto#L80

Are we sure this is going to work for more recent versions?

Reply:
IIRC the cookie is still in plain text

Reply:
Good point @fpj. As far as I know, this PR works with the defined version of Bookkeeper 4.14, which is the one we support today. In the case that the format changes in the future, we can address that in the future Bookkeeper upgrade and we can try to keep the change backwards compatible. Does it sound reasonable?

Reply:
I'll try to confirm in the code that it is writing text, but if anyone knows and wants to share in the meanwhile, then please do it.

The cookie layout version is 4 already, so it is an indication that it must be writing in plain text if we are able to read it now:

https://github.com/apache/bookkeeper/blob/branch-4.14/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java#L69

Reply:
It is fine from the write code path in Cookie.java:

https://github.com/apache/bookkeeper/blob/branch-4.14/bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Cookie.java#L223